cli/flox/doc/manifest.toml.md

```diff
@@ -253,67 +253,107 @@ DB_URL = "http://localhost:2000"
 SERVER_PORT = "3000"
 ```
 
-## `[profile]`
+## `[hook]`
 
-The `[profile]` section of the manifest allows you to specify scripts that will
-be *sourced* by your shell immediately after activating the environment.
-These scripts can be used to perform additional setup for a consumer of the
-environment e.g. spawning a background process, dynamically setting environment
-variables, or creating files and directories.  The environment variables
-declared in the `[vars]` section can be used in the `profile` scripts.
+Scripts in the `[hook]` section
+are useful for performing initialization
+in a predictable bash shell environment.
 
-```toml
-[profile]
-common = """
-    export SERVER_URL="http://localhost:$SERVER_PORT"
-"""
+### `on-activate`
 
-bash = """
-    export MYSHELL="bash"
-"""
+The `on-activate` script is invoked only once upon activation
+and can be useful for spawning processes,
+dynamically setting environment variables,
+and creating files and directories
+to be used by the subsequent profile scripts, commands and shells.
 
-zsh = """
-    export MYSHELL="zsh"
-"""
-```
-
-The `profile.common` script is intended to be common setup that can be sourced
-by any shell, but it is your responsibility to make sure that the script is
-compatible with any shells that may consume the environment.  The `profile.bash`
-and `profile.zsh` scripts are sourced *after* the `profile.common` script, and
-are only sourced by the corresponding shell.  The shell-specific profile scripts
-are intended to contain any shell functions, aliases, variables, etc that could
-be specific to a user's shell.
+If Flox detects that the environment has not yet been activated,
+then it will source this script from a **bash** shell.
+This prevents it from being sourced in subsequent "in place" activations,
```

Comment:
blocking: needs some explanation of how environment variables are handled from the hook. How flox activate (with hook that sets VAR) && export VAR=something-else && eval "$(flox activate)" will reset VAR

Reply:
We've already commented that this user-facing text falls into the product realm for maintenance over time, and the previous comments were that it was too long, so in Graham's absence Ross and I met to boil it down to the most succinct description. Let's set a deadline for you to meet with Ross and add whatever you think is necessary here as my input has already been incorporated.